content/docs/rfcs/19/README.md

```diff
@@ -0,0 +1,101 @@
+---
+slug: 19
+title: 19/WAKU2-DIRECT-MESSAGE
+name: Waku v2 Direct Message
+status: raw
+editor: Franck Royer <franck@status.im>
+contributors:
+---
+
+This specification explains the Waku v2 Direct Message protocol which enables a peer to send a direct message to another peer using the Waku v2 network.
+
+# Send message only knowing recipient's Ethereum Public Key
+
+This section describes how to use the Waku v2 Direct Message protocol to send a message to another party using their Ethereum Public Key.
+In this section, we assume that Alice sends a direct message to Bob.
+
+
+## Variables
+
+Here are the variables used in the protocol and their definition:
+
+- `A` is Alice's Ethereum root HD public key,
+- `B` is Bob's Ethereum root HD public key,
+- `a` is the private key of `A`, and is only known by Alice,
+- `b` is the private key of `B`, and is only known by Bob.
+
+## Design Requirements
+
+The proposed protocol MUST adhere to the following design requirements:
+
+a. Alice knows Bob's Ethereum root HD public key `B`,
+b. Alice wants to send message `M` to Bob using waku,
+c. Bob SHOULD be able to get `M` using [13/WAKU2-STORE](/spec/13), when querying a store node that hosts `M`,
+d. Bob MUST recognize he is `M`'s recipient when relaying it via [11/WAKU2-RELAY](/spec/11),
+e. Carole MUST NOT be able to read `M`'s content even if she is storing it or relaying it.
+
+## Steps
+
+1. Alice MUST derive Bob's waku public Key `Bw` from `B`,
+2. Alice SHOULD derive her own waku public key `Aw` from `A`,
+3. Alice creates `M'` which MUST contain `M` and `Aw`, it MAY contain `A`,
+4. Alice encrypts `M'` using `Bw`, resulting in `m'`, TODO: define encryption method,
+5. Alice creates waku message `Mw` with
+   `payload` `m'` and
+   `contentTopic` `/waku/2/direct-message/eth-pubkey/Bw/proto`,
+   with `Bw` in hex format (`0xAb1..`),
+6. Alice publishes `Mw` via [11/WAKU2-RELAY](/spec/11),
+7. Bob captures messages received messages via [11/WAKU2-RELAY](/spec/11) that have `contentTopic` `/waku/2/direct-message/eth-pubkey/Bw/proto`,
+8. Bob queries [13/WAKU2-STORE](/spec/13) with `contentTopics` set to `["/waku/2/direct-message/eth-pubkey/Bw/proto"]`,
+9. When retrieving `Mw` Bob derives `bw` from `b`,
+10. Bob uses `bw` to decrypt message `Mw`, he learns `m` and `Aw`,
+11. Bob replies to Alice in the same manner, setting the `contentTopic` to `/waku/2/direct-message/eth-pubkey/Aw/proto`.
+
+### Derivation
+
+Public parent key (`B`) to public child key (`Bw`) derivation is only possible with non-hardened paths [\[1\]](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki).
+
+TODO: Investigate commonly used derivation path to decide on one.
+
+### Reply
+
+To ascertain the fact that Alice receives Bob's reply, she could include connection details such as her peer id and multiaddress in the message.
```

Comment:
Ditto, inclusion of peerId is against anonymity and reveals both ends of communication